// ngx-webstorage/esm5/lib/strategies/baseSyncStorage.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { of, Subject } from 'rxjs';
import { CompatHelper } from '../helpers/compat';
/**
 * @abstract
 */
var /**
 * @abstract
 */
BaseSyncStorageStrategy = /** @class */ (function () {
    function BaseSyncStorageStrategy(storage, cache) {
        this.storage = storage;
        this.cache = cache;
        this.keyChanges = new Subject();
    }
    Object.defineProperty(BaseSyncStorageStrategy.prototype, "isAvailable", {
        get: /**
         * @return {?}
         */
        function () {
            if (this._isAvailable === undefined)
                this._isAvailable = CompatHelper.isStorageAvailable(this.storage);
            return this._isAvailable;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} key
     * @return {?}
     */
    BaseSyncStorageStrategy.prototype.get = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        /** @type {?} */
        var data = this.cache.get(this.name, key);
        if (data !== undefined)
            return of(data);
        try {
            /** @type {?} */
            var item = this.storage.getItem(key);
            if (item !== null) {
                data = JSON.parse(item);
                this.cache.set(this.name, key, data);
            }
        }
        catch (err) {
            console.warn(err);
        }
        return of(data);
    };
    /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    BaseSyncStorageStrategy.prototype.set = /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    function (key, value) {
        /** @type {?} */
        var data = JSON.stringify(value);
        this.storage.setItem(key, data);
        this.cache.set(this.name, key, value);
        this.keyChanges.next(key);
        return of(value);
    };
    /**
     * @param {?} key
     * @return {?}
     */
    BaseSyncStorageStrategy.prototype.del = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        this.storage.removeItem(key);
        this.cache.del(this.name, key);
        this.keyChanges.next(key);
        return of(null);
    };
    /**
     * @return {?}
     */
    BaseSyncStorageStrategy.prototype.clear = /**
     * @return {?}
     */
    function () {
        this.storage.clear();
        this.cache.clear(this.name);
        this.keyChanges.next(null);
        return of(null);
    };
    return BaseSyncStorageStrategy;
}());
/**
 * @abstract
 */
export { BaseSyncStorageStrategy };
if (false) {
    /** @type {?} */
    BaseSyncStorageStrategy.prototype.keyChanges;
    /** @type {?} */
    BaseSyncStorageStrategy.prototype.name;
    /**
     * @type {?}
     * @protected
     */
    BaseSyncStorageStrategy.prototype._isAvailable;
    /**
     * @type {?}
     * @protected
     */
    BaseSyncStorageStrategy.prototype.storage;
    /**
     * @type {?}
     * @protected
     */
    BaseSyncStorageStrategy.prototype.cache;
}
