// ngx-webstorage/esm2015/public_api.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/*
 * Public API Surface of ngx-webstorage
 */
export { StorageStrategies } from './lib/constants/strategy';
export { CompatHelper } from './lib/helpers/compat';
export { SyncStorage } from './lib/core/templates/syncStorage';
export { AsyncStorage } from './lib/core/templates/asyncStorage';
export { StrategyCacheService } from './lib/core/strategyCache';
export { LOCAL_STORAGE, SESSION_STORAGE } from './lib/core/nativeStorage';
export { STORAGE_STRATEGIES } from './lib/strategies/index';
export { LocalStorageStrategy } from './lib/strategies/localStorage';
export { SessionStorageStrategy } from './lib/strategies/sessionStorage';
export { InMemoryStorageStrategy } from './lib/strategies/inMemory';
export { StorageStrategyStubName, StorageStrategyStub } from './stubs/storageStrategy.stub';
export { StorageStub } from './stubs/storage.stub';
export { InvalidStrategyError, StrategyIndex } from './lib/services/strategyIndex';
export { LocalStorageService } from './lib/services/localStorage';
export { SessionStorageService } from './lib/services/sessionStorage';
export {} from './lib/core/interfaces/storageStrategy';
export { LocalStorage, SessionStorage } from './lib/decorators';
export { appInit, LIB_CONFIG, NgxWebstorageModule } from './lib/module';
