// ngx-webstorage/esm2015/stubs/storage.stub.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
export class StorageStub {
    constructor() {
        this.store = {};
    }
    /**
     * @return {?}
     */
    get length() {
        return Object.keys(this.store).length;
    }
    /**
     * @return {?}
     */
    clear() {
        this.store = {};
    }
    /**
     * @param {?} key
     * @return {?}
     */
    getItem(key) {
        return this.store[key] || null;
    }
    /**
     * @param {?} index
     * @return {?}
     */
    key(index) {
        return Object.keys(this.store)[index];
    }
    /**
     * @param {?} key
     * @return {?}
     */
    removeItem(key) {
        delete this.store[key];
    }
    /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    setItem(key, value) {
        this.store[key] = value;
    }
}
if (false) {
    /** @type {?} */
    StorageStub.prototype.store;
    /* Skipping unhandled member: [name: string]: any;*/
}
