// ngx-webstorage/esm2015/stubs/storageStrategy.stub.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { of, Subject } from 'rxjs';
import { Optional } from '@angular/core';
/** @type {?} */
export const StorageStrategyStubName = 'stub_strategy';
export class StorageStrategyStub {
    /**
     * @param {?=} name
     */
    constructor(name) {
        this.keyChanges = new Subject();
        this.store = {};
        this._available = true;
        this.name = name || StorageStrategyStubName;
    }
    /**
     * @return {?}
     */
    get isAvailable() {
        return this._available;
    }
    /**
     * @param {?} key
     * @return {?}
     */
    get(key) {
        return of(this.store[key]);
    }
    /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    set(key, value) {
        this.store[key] = value;
        this.keyChanges.next(key);
        return of(value);
    }
    /**
     * @param {?} key
     * @return {?}
     */
    del(key) {
        delete this.store[key];
        this.keyChanges.next(key);
        return of(null);
    }
    /**
     * @return {?}
     */
    clear() {
        this.store = {};
        this.keyChanges.next(null);
        return of(null);
    }
}
/** @nocollapse */
StorageStrategyStub.ctorParameters = () => [
    { type: String, decorators: [{ type: Optional }] }
];
if (false) {
    /** @type {?} */
    StorageStrategyStub.prototype.keyChanges;
    /** @type {?} */
    StorageStrategyStub.prototype.store;
    /** @type {?} */
    StorageStrategyStub.prototype._available;
    /** @type {?} */
    StorageStrategyStub.prototype.name;
}
