// ngx-webstorage/esm5/lib/services/localStorage.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { SyncStorage } from '../core/templates/syncStorage';
import { StrategyIndex } from './strategyIndex';
import { StorageStrategies } from '../constants/strategy';
var LocalStorageService = /** @class */ (function (_super) {
    tslib_1.__extends(LocalStorageService, _super);
    function LocalStorageService() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return LocalStorageService;
}(SyncStorage));
export { LocalStorageService };
/**
 * @param {?} index
 * @return {?}
 */
export function buildService(index) {
    /** @type {?} */
    var strategy = index.indexStrategy(StorageStrategies.Local);
    return new SyncStorage(strategy);
}
/** @type {?} */
export var LocalStorageServiceProvider = {
    provide: LocalStorageService,
    useFactory: buildService,
    deps: [StrategyIndex]
};
