// ngx-webstorage/esm5/lib/services/sessionStorage.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { SyncStorage } from '../core/templates/syncStorage';
import { StrategyIndex } from './strategyIndex';
import { StorageStrategies } from '../constants/strategy';
var SessionStorageService = /** @class */ (function (_super) {
    tslib_1.__extends(SessionStorageService, _super);
    function SessionStorageService() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return SessionStorageService;
}(SyncStorage));
export { SessionStorageService };
/**
 * @param {?} index
 * @return {?}
 */
export function buildService(index) {
    /** @type {?} */
    var strategy = index.indexStrategy(StorageStrategies.Session);
    return new SyncStorage(strategy);
}
/** @type {?} */
export var SessionStorageServiceProvider = {
    provide: SessionStorageService,
    useFactory: buildService,
    deps: [StrategyIndex]
};
