// ngx-webstorage/esm2015/lib/core/templates/syncStorage.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { noop } from '../../helpers/noop';
import { StorageKeyManager } from '../../helpers/storageKeyManager';
import { distinctUntilChanged, filter, shareReplay, switchMap } from 'rxjs/operators';
export class SyncStorage {
    /**
     * @param {?} strategy
     */
    constructor(strategy) {
        this.strategy = strategy;
    }
    /**
     * @param {?} key
     * @return {?}
     */
    retrieve(key) {
        /** @type {?} */
        let value;
        this.strategy.get(StorageKeyManager.normalize(key)).subscribe((/**
         * @param {?} result
         * @return {?}
         */
        (result) => value = typeof result === 'undefined' ? null : result));
        return value;
    }
    /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    store(key, value) {
        this.strategy.set(StorageKeyManager.normalize(key), value).subscribe(noop);
        return value;
    }
    /**
     * @param {?=} key
     * @return {?}
     */
    clear(key) {
        if (key !== undefined)
            this.strategy.del(StorageKeyManager.normalize(key)).subscribe(noop);
        else
            this.strategy.clear().subscribe(noop);
    }
    /**
     * @return {?}
     */
    getStrategyName() { return this.strategy.name; }
    /**
     * @param {?} key
     * @return {?}
     */
    observe(key) {
        key = StorageKeyManager.normalize(key);
        return this.strategy.keyChanges.pipe(filter((/**
         * @param {?} changed
         * @return {?}
         */
        (changed) => changed === null || changed === key)), switchMap((/**
         * @return {?}
         */
        () => this.strategy.get(key))), distinctUntilChanged(), shareReplay());
    }
}
if (false) {
    /**
     * @type {?}
     * @protected
     */
    SyncStorage.prototype.strategy;
}
