// ngx-webstorage/esm2015/lib/helpers/storageKeyManager.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { DefaultIsCaseSensitive, DefaultPrefix, DefaultSeparator } from '../constants/config';
export class StorageKeyManager {
    /**
     * @param {?} raw
     * @return {?}
     */
    static normalize(raw) {
        raw = StorageKeyManager.isCaseSensitive ? raw : raw.toLowerCase();
        return `${StorageKeyManager.prefix}${StorageKeyManager.separator}${raw}`;
    }
    /**
     * @param {?} key
     * @return {?}
     */
    static isNormalizedKey(key) {
        return key.indexOf(StorageKeyManager.prefix + StorageKeyManager.separator) === 0;
    }
    /**
     * @param {?} prefix
     * @return {?}
     */
    static setPrefix(prefix) {
        StorageKeyManager.prefix = prefix;
    }
    /**
     * @param {?} separator
     * @return {?}
     */
    static setSeparator(separator) {
        StorageKeyManager.separator = separator;
    }
    /**
     * @param {?} enable
     * @return {?}
     */
    static setCaseSensitive(enable) {
        StorageKeyManager.isCaseSensitive = enable;
    }
    /**
     * @param {?} config
     * @return {?}
     */
    static consumeConfiguration(config) {
        if ('prefix' in config)
            this.setPrefix(config.prefix);
        if ('separator' in config)
            this.setSeparator(config.separator);
        if ('caseSensitive' in config)
            this.setCaseSensitive(config.caseSensitive);
    }
}
StorageKeyManager.prefix = DefaultPrefix;
StorageKeyManager.separator = DefaultSeparator;
StorageKeyManager.isCaseSensitive = DefaultIsCaseSensitive;
if (false) {
    /** @type {?} */
    StorageKeyManager.prefix;
    /** @type {?} */
    StorageKeyManager.separator;
    /** @type {?} */
    StorageKeyManager.isCaseSensitive;
}
