// ngx-webstorage/esm5/stubs/storage.stub.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var StorageStub = /** @class */ (function () {
    function StorageStub() {
        this.store = {};
    }
    Object.defineProperty(StorageStub.prototype, "length", {
        get: /**
         * @return {?}
         */
        function () {
            return Object.keys(this.store).length;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    StorageStub.prototype.clear = /**
     * @return {?}
     */
    function () {
        this.store = {};
    };
    /**
     * @param {?} key
     * @return {?}
     */
    StorageStub.prototype.getItem = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        return this.store[key] || null;
    };
    /**
     * @param {?} index
     * @return {?}
     */
    StorageStub.prototype.key = /**
     * @param {?} index
     * @return {?}
     */
    function (index) {
        return Object.keys(this.store)[index];
    };
    /**
     * @param {?} key
     * @return {?}
     */
    StorageStub.prototype.removeItem = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        delete this.store[key];
    };
    /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    StorageStub.prototype.setItem = /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    function (key, value) {
        this.store[key] = value;
    };
    return StorageStub;
}());
export { StorageStub };
if (false) {
    /** @type {?} */
    StorageStub.prototype.store;
    /* Skipping unhandled member: [name: string]: any;*/
}
