// ngx-webstorage/esm5/stubs/storageStrategy.stub.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { of, Subject } from 'rxjs';
import { Optional } from '@angular/core';
/** @type {?} */
export var StorageStrategyStubName = 'stub_strategy';
var StorageStrategyStub = /** @class */ (function () {
    function StorageStrategyStub(name) {
        this.keyChanges = new Subject();
        this.store = {};
        this._available = true;
        this.name = name || StorageStrategyStubName;
    }
    Object.defineProperty(StorageStrategyStub.prototype, "isAvailable", {
        get: /**
         * @return {?}
         */
        function () {
            return this._available;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} key
     * @return {?}
     */
    StorageStrategyStub.prototype.get = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        return of(this.store[key]);
    };
    /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    StorageStrategyStub.prototype.set = /**
     * @param {?} key
     * @param {?} value
     * @return {?}
     */
    function (key, value) {
        this.store[key] = value;
        this.keyChanges.next(key);
        return of(value);
    };
    /**
     * @param {?} key
     * @return {?}
     */
    StorageStrategyStub.prototype.del = /**
     * @param {?} key
     * @return {?}
     */
    function (key) {
        delete this.store[key];
        this.keyChanges.next(key);
        return of(null);
    };
    /**
     * @return {?}
     */
    StorageStrategyStub.prototype.clear = /**
     * @return {?}
     */
    function () {
        this.store = {};
        this.keyChanges.next(null);
        return of(null);
    };
    /** @nocollapse */
    StorageStrategyStub.ctorParameters = function () { return [
        { type: String, decorators: [{ type: Optional }] }
    ]; };
    return StorageStrategyStub;
}());
export { StorageStrategyStub };
if (false) {
    /** @type {?} */
    StorageStrategyStub.prototype.keyChanges;
    /** @type {?} */
    StorageStrategyStub.prototype.store;
    /** @type {?} */
    StorageStrategyStub.prototype._available;
    /** @type {?} */
    StorageStrategyStub.prototype.name;
}
